### PR TITLE
bindings(python): add `RewardsParameters` class

### LINKED ISSUE #1290: Add RewardsParameters to ProtocolParameters - Python

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

Introduces the `RewardsParameters` class and adds it as a field to the `ProtocolParameters`.

## Links to any relevant issues

Closes #1290

### CODE REVIEW

**Review by DaughterOfMars — DISMISSED** [review]

File: bindings/python/iota_sdk/types/node_info.py [hunk 1]
```diff
@@ -189,6 +189,29 @@ class ManaStructure:
...
+    Attributes:
+        validator_blocks_per_slot: The number of validation blocks that should be issued by a selected validator per slot during its epoch duties.
+        profit_margin_exponent: Used for shift operation during calculation of profit margin.
+        bootstrapping_duration: The length of the bootstrapping phase in epochs.
+        mana_share_coefficient: The coefficient used for calculation of initial rewards.
+        decay_balancing_constant_exponent: The exponent used for calculation of the initial reward.
+        decay_balancing_constant: An integer approximation which is calculated using the `decay_balancing_constant_exponent`.
+        pool_coefficient_exponent: The exponent used for shifting operation during the pool rewards calculations.
+    """
+    validator_blocks_per_slot: int
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/node_info.py]: https://github.com/iotaledger/tips-draft/pull/25#issuecomment-1755703335 :shrug:

File: bindings/python/iota_sdk/types/node_info.py [hunk 2]
```diff
@@ -189,6 +189,29 @@ class ManaStructure:
...
+    profit_margin_exponent: int
+    bootstrapping_duration: int
+    mana_share_coefficient: int
+    decay_balancing_constant_exponent: int
+    decay_balancing_constant: int
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/node_info.py]:
```suggestion
    mana_share_coefficient: int = field(metadata=config(
        encoder=str
    ))
    decay_balancing_constant_exponent: int
    decay_balancing_constant: int = field(metadata=config(
        encoder=str
    ))
```

Comment (Alex6323) [reply bindings/python/iota_sdk/types/node_info.py]: for the bootstrapping duration as well, right?

Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/node_info.py]: No, EpochIndex is also u32 now

Comment (Alex6323) [reply bindings/python/iota_sdk/types/node_info.py]: okay

File: bindings/python/iota_sdk/types/node_info.py (hunk 1 above)
Comment (Alex6323) [reply bindings/python/iota_sdk/types/node_info.py]: Fixed

**Review by DaughterOfMars — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**